packages/web-components/fast-foundation/src/combobox/combobox.spec.ts

```diff
@@ -150,6 +150,93 @@ describe("Combobox", () => {
         await disconnect();
     });
 
+    it("should NOT emit a 'change' event when the user presses Enter without changing value", async () => {
+        const { element, connect, disconnect } = await setup();
+
+        await connect();
+
+        element.value = "two";
+
+        const event = new KeyboardEvent("keydown", {
+            key: keyEnter,
+        } as KeyboardEventInit);
+
+        const wasChanged = await Promise.race([
+            new Promise(resolve => {
+                element.addEventListener("change", () => resolve(true));
+                element.dispatchEvent(event);
+            }),
+            DOM.nextUpdate().then(() => false),
+        ]);
+
+        expect(wasChanged).to.be.false;
+
+        await disconnect();
+
+    });
+
+    it("should update value to entered non-option value after selecting an option value", async () => {
+        const { element, connect, disconnect } = await setup();
+
+        await connect();
+
+        element.value = "two";
+
+        const enterEvent = new KeyboardEvent("keydown", {
+            key: keyEnter,
+        } as KeyboardEventInit);
+
+        const wasChanged = await Promise.race([
+            new Promise(resolve => {
+                element.addEventListener("change", () => resolve(true));
+
+                // fake a key entered value
+                (element as Combobox).control.value = 'a';
+                (element as Combobox).control.dispatchEvent(new InputEvent('input', { data: 'a', inputType: 'insertText' }));
+                DOM.nextUpdate();
+
+                element.dispatchEvent(enterEvent);
+            }),
+            DOM.nextUpdate().then(() => false),
+        ]);
+
+        expect(wasChanged).to.be.true;
+        expect((element as Combobox).value).to.equal('a');
+
+        await disconnect();
+
+    });
+
+    it("should emit a 'change' event when the user clicks away after selecting option in dropdown", async () => {
+        const { element, connect, disconnect } = await setup();
+
+        await connect();
+
+        element.value = "two";
+        element.click(); // open dropdown
+
+        const keyDownEvent = new KeyboardEvent("keydown", {
+            key: keyArrowDown,
+        } as KeyboardEventInit);
+
+        const wasChanged = await Promise.race([
+            new Promise(resolve => {
+                element.addEventListener("change", () => resolve(true));
+
+                element.dispatchEvent(keyDownEvent);
+                DOM.nextUpdate();
```

Comment:
Hmmm...I don't think this is valid, as I'm not awaiting this call to `nextUpdate()`, and if I did, I would presume this could introduce intermittency to the test. I'll think about another way to write this test.

Reply:
Simple re-org of code addressed this.